contracts/GenericLender/GenericAave.sol

```diff
@@ -74,24 +111,118 @@ contract GenericAave is GenericLenderBase {
+        (uint256 availableLiquidity, uint256 totalStableDebt, uint256 totalVariableDebt, , , , uint256 averageStableBorrowRate, , , ) =
+            protocolDataProvider.getReserveData(address(want));
+
+        uint256 newLiquidity = availableLiquidity.add(extraAmount);
+
+        (, , , , uint256 reserveFactor, , , , , ) = protocolDataProvider.getReserveConfigurationData(address(want));
+
+        (uint256 newLiquidityRate, , ) =
+            IReserveInterestRateStrategy(reserveData.interestRateStrategyAddress).calculateInterestRates(
+                address(want),
+                newLiquidity,
+                totalStableDebt,
+                totalVariableDebt,
+                averageStableBorrowRate,
+                reserveFactor
+            );
+
+        uint256 incentivesRate = _incentivesRate(newLiquidity.add(totalStableDebt).add(totalVariableDebt)); // total supplied liquidity in Aave v2
+        return newLiquidityRate.div(1e9).add(incentivesRate); // divided by 1e9 to go from Ray to Wad
+    }
+
+    function hasAssets() external view override returns (bool) {
+        return aToken.balanceOf(address(this)) > 0;
+    }
+
+    // Only for incentivised aTokens
+    // this is a manual trigger to claim rewards once each 10 days
+    // only callable if the token is incentivised by Aave Governance (_checkCooldown returns true)
+    function harvest() external {
+        if(_checkCooldown()) {
+            // redeem AAVE from stkAave
+            uint256 stkAaveBalance = IERC20(address(stkAave)).balanceOf(address(this));
+            if(stkAaveBalance > 0) {
+                stkAave.redeem(address(this), stkAaveBalance);
+            }
+
+            // sell AAVE for want
+            uint256 aaveBalance = IERC20(AAVE).balanceOf(address(this));
+            _sellAAVEForWant(aaveBalance);
+            
+            // deposit want in lending protocol 
+            uint256 balance = want.balanceOf(address(this));
+            if(balance > 0) {
+                _deposit(balance);
+            }
+
+            // claim rewards
+            address[] memory assets = new address[](1);
+            assets[0] = address(aToken);
+            uint256 pendingRewards = _incentivesController().getRewardsBalance(assets, address(this));
+            if(pendingRewards > 0) {
+                _incentivesController().claimRewards(assets, pendingRewards, address(this));
+            }
+
+            // request start of cooldown period
+            if(IERC20(address(stkAave)).balanceOf(address(this)) > 0) {
+                stkAave.cooldown();
+            }
+        }
+    }
+
+    function harvestTrigger() external view returns (bool) {
+        return _checkCooldown();
+    }
+
+    function _initialize(IAToken _aToken, bool _isIncentivised) internal {
+        require(address(aToken) == address(0), "GenericAave already initialized");
+    
+        isIncentivised = _isIncentivised;
```

Comment:
should this add a require similar to this one if the var is true?

`require(address(aToken.getIncentivesController()) != address(0), "!aToken does not have incentives controller set up");`

Reply:
added